assignments/homework/pcosta/497HW1.py

```diff
@@ -0,0 +1,70 @@
+import numpy as np
+import matplotlib.pyplot as plt
+from astropy import units as u
+from astropy.units import imperial
+from astropy import constants as const
+from astropy.time import Time
+from astropy.coordinates import SkyCoord
+from astropy.coordinates import EarthLocation, AltAz
+from random import *
+from astropy.coordinates import FK5
+
+#Units Notebook Homework
+#1-Unit Conversions
+velocities = np.array([2.3, 3.5, 6.0, 10.5])*u.m/u.s
+convertedVelocities = velocities.to(imperial.mi/u.h)
+print(convertedVelocities)
+
+#2-Using Equivalencies
+specDensity1 = (12 * u.mJy).to(u.erg / u.cm**2 / u.s / u.Hz)
+print(specDensity1)
+
+specDensity2 = (12 * u.mJy).to(u.W / u.m**2 / u.Hz)
+print(specDensity2)
+
+#3-Plotting the Solar System Potential
+def V(r):
+    return -1 * const.G * const.M_sun / r
+R = ((np.arange(1, 50, 0.5))*u.AU)
+plt.plot(R, V(R).to(u.mJ/u.kg))
+plt.title("Gravitational Potential")
+plt.xlabel("Radius (AU)")
+plt.ylabel("Potential (MJ/Kg)")
+plt.show()
+
+#Times Notebook Homework
+t1 = Time.now()
+print(t1)
+Start = Time("2021-04-02 12:00:00")
+print((t1-Start).to(u.min))
+
+#Coordinates Notebook
+#1.
+m1 = SkyCoord.from_name('M1')
+M1 = m1.transform_to('galactic')
+print(M1)
+
+#2.
+tauri = SkyCoord.from_name('T Tauri')
+seattle = EarthLocation(lat=47.6062*u.deg, lon=-122.3321*u.deg, height=0*u.m)
+utc_time = Time.now()
+m1_altaz = m1.transform_to(AltAz(obstime=utc_time, location=seattle))
+tauri_altaz = tauri.transform_to(AltAz(obstime=utc_time, location=seattle))
+separation = (tauri_altaz.az).deg*u.deg - (m1_altaz.az).deg*u.deg
+print(separation)
```

Comment:
This solution is not correct. You calculate the separation in azimuth only rather than the separation in the sky coordinate system, which takes into account spherical geometry as well as both coordinates. So for the solution there is no need to transform to AltAz frame, you would only need this one line:

```
separation = tauri.separation(m1).to(u.deg)
```